### ISSUE TITLE
Regression in 0.0.71: generic TypedDict union alias in stub leaks unspecialized TypeVar

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#4255]
### Summary

A generic union alias defined in a `.pyi` file loses its concrete type argument when consumed from Python code. In ty 0.0.69, this imprecisely resolves to `Unknown` but does not produce an error; in 0.0.71, the alias’s internal `TypeVar` escapes into the consumer and causes a false-positive `unresolved-attribute` diagnostic.

**`events.pyi`:**

```python
from collections.abc import Iterator
from typing import Generic, Literal, TypeVar, TypedDict, Union

T = TypeVar("T")


class ObjectEvent(TypedDict, Generic[T]):
    type: Literal["ADDED"]
    object: T


class BookmarkEvent(TypedDict):
    type: Literal["BOOKMARK"]
    object: object


DecodedEvent = Union[ObjectEvent[T], BookmarkEvent, None]


def events() -> Iterator[DecodedEvent[int]]: ...
```

**`example.py`:**

```python
from typing import reveal_type

from events import events


for event in events():
    if event is None or event["type"] == "BOOKMARK":
        continue

    value = event["object"]
    reveal_type(value)
    print(value.bit_length())
```

**Expected:** `value` should be inferred as `int`, because `DecodedEvent[int]` specializes `ObjectEvent[T]` to `ObjectEvent[int]`, and the other union members have been excluded.

**ty 0.0.69:**

```text
example.py:9:17: info[revealed-type] Revealed type: `Unknown`
```

The inference is incomplete, but checking succeeds.

**ty 0.0.71:**

```text
example.py:9:17: info[revealed-type] Revealed type: `T@DecodedEvent`
example.py:10:11: error[unresolved-attribute] Object of type
`T@DecodedEvent` has no attribute `bit_length`
```

**Versions tested:**

```text
ty 0.0.69 (1dca8443e 2026-08-06)
ty 0.0.71 (683e22331 2026-08-12)
```

This surfaced with Kubernetes watch stubs, where `_DecodedWatchItem = Union[_ObjectEvent[T], _BookmarkEvent, None]` should preserve the concrete event model through `_DecodedWatchItem[CoreV1Event]`. An explicit `isinstance` check works around the issue, but the concrete type should already be known.

### Version

0.0.71
